### PR TITLE
[EMB-314] Display a banner from the API

### PR DESCRIPTION
**Pull request by aaxelb**
## Purpose
When a banner is created in the admin, display it on the landing page (home/dashboard).

## Summary of Changes
* Add `banner` model/adapter/serializer, for fetching `/_/banners/current/`
* Add `scheduled-banner` component, to display that banner on the dashboard and home page

## Side Effects / Testing Notes
* Should have exactly the same behavior as the banner in registries and preprints

## Ticket

https://openscience.atlassian.net/browse/EMB-314

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: app/adapters/banner.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import OsfAdapter from './osf-adapter';
+
+export default class BannerAdapter extends OsfAdapter {
+    namespace = '_';
```
Comment (jamescdavis) [inline app/adapters/banner.ts]: 👏

File: app/serializers/banner.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+export default class BannerAdapter extends OsfSerializer {
+    normalizeSingleResponse(
+        store: DS.Store,
+        primaryModelClass: any,
+        payload: any,
+        id: string,
+        requestType: string,
+    ): any {
+        // The banner payload currently has an empty ID
```
Comment (jamescdavis) [inline app/serializers/banner.ts]: Maybe we should "correct" this on the back-end?

File: lib/osf-components/addon/components/scheduled-banner/x-maybe-link/template.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{{#if href}}
+    <a
+        href={{href}}
+        onclick={{action 'click' 'link' (concat 'Banner - ' banner.name) target=analytics}}
+        target="_blank"
+        rel="noopener"
```
Comment (jamescdavis) [inline lib/osf-components/addon/components/scheduled-banner/x-maybe-link/template.hbs]: 🔒

File: app/serializers/banner.ts (hunk 2 above)
Comment (aaxelb) [reply app/serializers/banner.ts]: Sounds worth a ticket, but might require updating the other apps that are already using this API.

**Pull request merged**